drivers/pwm/pwm_numaker.c

```diff
@@ -0,0 +1,574 @@
+/*
+ * Copyright (c) 2023 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nuvoton_numaker_pwm
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/reset.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/pwm.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/clock_control_numaker.h>
+#include <zephyr/logging/log.h>
+#include <soc.h>
+#include <NuMicro.h>
+
+LOG_MODULE_REGISTER(pwm_numaker, LOG_LEVEL_ERR);
+
+/* 11-bit prescaler in Numaker EPWM modules */
+#define NUMAKER_PWM_MAX_PRESCALER (1UL << (11))
+#define NUMAKER_PWM_CHANNEL_COUNT 6
+#define NUMAKER_PWM_RELOAD_CNT    (0xFFFFU)
+
+/* Device config */
+struct pwm_numaker_config {
+	/* EPWM base address */
+	EPWM_T *epwm;
+	uint32_t prescale;
+	const struct reset_dt_spec reset;
+	/* clock configuration */
+	uint32_t clk_modidx;
+	uint32_t clk_src;
+	uint32_t clk_div;
+	const struct device *clk_dev;
+	const struct pinctrl_dev_config *pincfg;
+	void (*irq_config_func)(const struct device *dev);
+};
+
+struct pwm_numaker_capture_data {
+	pwm_capture_callback_handler_t callback;
+	void *user_data;
+	/* Only support either one of PWM_CAPTURE_TYPE_PULSE, PWM_CAPTURE_TYPE_PERIOD */
+	bool pulse_capture;
+	bool single_mode;
+	bool is_busy;
+	uint32_t curr_edge_mode;
+	uint32_t next_edge_mode;
+};
+
+/* Driver context/data */
+struct pwm_numaker_data {
+	uint32_t clock_freq;
+	uint32_t cycles_per_sec;
+#ifdef CONFIG_PWM_CAPTURE
+	uint32_t overflows;
+	struct pwm_numaker_capture_data capture[NUMAKER_PWM_CHANNEL_COUNT];
+#endif /* CONFIG_PWM_CAPTURE */
+};
+
+static void pwm_numaker_configure(const struct device *dev)
+{
+	const struct pwm_numaker_config *cfg = dev->config;
+	EPWM_T *epwm = cfg->epwm;
+
+	/* Disable EPWM channel 0~5 before config */
+	EPWM_ForceStop(epwm, 0x3F);
+
+	/* Set EPWM default normal polarity as inverse disabled */
+	epwm->POLCTL &= ~(0x3F << EPWM_POLCTL_PINV0_Pos);
+}
+
+/* PWM api functions */
+static int pwm_numaker_set_cycles(const struct device *dev, uint32_t channel,
+				  uint32_t period_cycles, uint32_t pulse_cycles, pwm_flags_t flags)
+{
+	const struct pwm_numaker_config *cfg = dev->config;
+	struct pwm_numaker_data *data = dev->data;
+	EPWM_T *epwm = cfg->epwm;
+	uint32_t channel_mask = (1 << channel);
+
+	LOG_DBG("===>[%s] Channel=0x%x, CAPIEN=0x%x, CAPIF=0x%x", __func__, channel, epwm->CAPIEN,
```

Comment:
I'd suggest dropping decorators like the `===>` here. Fine while you are debugging this one but if we start having them through the whole code base it'd all become a giant mess if you enable debugging. Just log what you need to log. Also no need for `__func__`, `LOG_DBG` already logs it.

Same for all the other `LOG_DBG`